functions/_tide_item_kubectl.fish

```diff
@@ -1,4 +1,4 @@
 function _tide_item_kubectl
-    kubectl config view --minify --output 'jsonpath={.current-context}/{..namespace}' 2>/dev/null | read -l context &&
+    kubectl config view --minify --output 'jsonpath={.current-context}/{..namespace}' 2>/dev/null | string shorten -"$tide_kubectl_truncation_strategy"m"$tide_kubectl_truncation_length" | read -l context &&
```

Comment:
This doesn't work in MacOS 14.2.1.
Maybe use AWK instead of string? 
```
function _tide_item_kubectl
    set context (kubectl config view --minify --output 'jsonpath={.current-context}/{..namespace}' 2>/dev/null | \
                 awk -v len="$tide_kubectl_truncation_length" '{print substr($0, 1, len)}')
    if set -q context[1]
        _tide_print_item kubectl $tide_kubectl_icon' ' (string replace -r '/(|default)$' '' $context)
    end
end
```

Reply:
@0xMH I am open to using awk but the snippet above returns the beginning and not the end of the string.

Example:

The command

```sh
kubectl config view --minify --output 'jsonpath={.current-context}/{..namespace}'
```

returns the untruncated string:

```
arn:aws:eks:<region>:<numbers>:cluster/<clustername>/<namespace>
```

The snippet above returns `arn:aws:eks:<region>:<some_numbers>` instead of `<clustername>/<namespace>` for a given string length.

Reply:
Now? 
```
function _tide_item_kubectl
    set context (kubectl config view --minify --output 'jsonpath={.current-context}/{..namespace}' 2>/dev/null | \
                 awk -v len="$tide_kubectl_truncation_length" '{print substr($0, length($0)-len+1)}')
    if set -q context[1]
        _tide_print_item kubectl $tide_kubectl_icon' ' (string replace -r '/(|default)$' '' $context)
    end
end
```

Reply:
Works

Reply:
what do you think @IlanCosman?